### PR TITLE
Add fast string hash

### PR DESCRIPTION
**Pull request by ibmibmibm**

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Review by dannypsnl — DISMISSED** [review]

File: lib/common/hash.cpp [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+  X -= (X >> 1) & 0x5555555555555555;
+  X = (X & 0x3333333333333333) + ((X >> 2) & 0x3333333333333333);
+  X = (X + (X >> 4)) & 0x0f0f0f0f0f0f0f0f;
+  X = (X * 0x0101010101010101) >> 56;
+  return static_cast<int>(X);
+#endif
+}
+
+std::array<uint64_t, 4> generate() noexcept {
+  std::array<uint64_t, 4> secret;
```
Comment (hydai) [inline lib/common/hash.cpp:145]: `Secret`

File: lib/common/hash.cpp [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  const std::array<uint8_t, 70> c = {
```
Comment (hydai) [inline lib/common/hash.cpp:146]: `C`

File: lib/common/hash.cpp [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+      0x0f, 0x17, 0x1b, 0x1d, 0x1e, 0x27, 0x2b, 0x2d, 0x2e, 0x33, 0x35, 0x36,
+      0x39, 0x3a, 0x3c, 0x47, 0x4b, 0x4d, 0x4e, 0x53, 0x55, 0x56, 0x59, 0x5a,
+      0x5c, 0x63, 0x65, 0x66, 0x69, 0x6a, 0x6c, 0x71, 0x72, 0x74, 0x78, 0x87,
+      0x8b, 0x8d, 0x8e, 0x93, 0x95, 0x96, 0x99, 0x9a, 0x9c, 0xa3, 0xa5, 0xa6,
+      0xa9, 0xaa, 0xac, 0xb1, 0xb2, 0xb4, 0xb8, 0xc3, 0xc5, 0xc6, 0xc9, 0xca,
+      0xcc, 0xd1, 0xd2, 0xd4, 0xd8, 0xe1, 0xe2, 0xe4, 0xe8, 0xf0};
+  std::uniform_int_distribution<uint64_t> Dist(
+      UINT64_C(0), static_cast<uint64_t>(c.size() - 1));
+  for (size_t i = 0; i < 4; i++) {
```
Comment (hydai) [inline lib/common/hash.cpp:155]: `I`

File: lib/common/hash.cpp [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+    bool ok;
```
Comment (hydai) [inline lib/common/hash.cpp:156]: `Ok`

File: lib/common/hash.cpp [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    do {
+      ok = true;
+      secret[i] = 0;
+      for (size_t j = 0; j < 64; j += 8) {
```
Comment (hydai) [inline lib/common/hash.cpp:160]: `J`

**Review by hydai — CHANGES_REQUESTED** [review]
Naming style issue

**Review by hydai — CHANGES_REQUESTED** [review]
Fedora rawhide failed. It seems like the LLVM C API we used before is deprecated in the latest version

**Comment by ibmibmibm** [comment]
> Fedora rawhide failed. It seems like the LLVM C API we used before is deprecated in the latest version

Fixing in #3809

**Review by hydai — APPROVED** [review]

**Pull request merged**